lib/spack/spack/test/util/editor.py

```diff
@@ -132,3 +132,17 @@ def assert_exec(exe, args):
         assert args == [vim_exe, '/path/to/file']
 
     ed.editor('/path/to/file', _exec_func=assert_exec)
+
+
+def test_no_editor():
+    if 'VISUAL' in os.environ:
+        del os.environ['VISUAL']
+    if 'EDITOR' in os.environ:
+        del os.environ['EDITOR']
+    os.environ['PATH'] = ''
+
+    def assert_exec(exe, args):
+        assert False
+
+    with pytest.raises(EnvironmentError):
```

Comment:
Here you can also try to match the message in the exception using a regex, if you think it would be worthwhile. Check for the `match=` argument in case.